apps/api-journeys/src/app/modules/event/textResponse/textResponse.resolver.ts

```
@@ -4,26 +4,59 @@ import { UseGuards } from '@nestjs/common'
 import { Args, Mutation, Resolver } from '@nestjs/graphql'
 import { GqlAuthGuard } from '@core/nest/gqlAuthGuard/GqlAuthGuard'
 import { CurrentUserId } from '@core/nest/decorators/CurrentUserId'
+import { UserInputError } from 'apollo-server'
 import {
   TextResponseSubmissionEvent,
   TextResponseSubmissionEventCreateInput
 } from '../../../__generated__/graphql'
 import { EventService } from '../event.service'
+import { BlockService } from '../../block/block.service'
+import { VisitorService } from '../../visitor/visitor.service'
 
 @Resolver('TextResponseSubmissionEvent')
 export class TextResponseSubmissionEventResolver {
-  constructor(private readonly eventService: EventService) {}
+  constructor(
+    private readonly eventService: EventService,
+    private readonly blockService: BlockService,
+    private readonly visitorService: VisitorService
+  ) {}
+
   @Mutation()
   @UseGuards(GqlAuthGuard)
   async textResponseSubmissionEventCreate(
     @CurrentUserId() userId: string,
     @Args('input') input: TextResponseSubmissionEventCreateInput
   ): Promise<TextResponseSubmissionEvent> {
+    const block: { journeyId: string } = await this.blockService.get(
+      input.blockId
+    )
+    const journeyId = block.journeyId
```

need to throw a UserInput Error here if no block.